sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -158,5 +159,22 @@ class DDLSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("SPARK-14129: alter table - rename") {
+    import testImplicits._
+
+    withTempPath { dir =>
+      val path = dir.getCanonicalPath
+      val df =
+        sparkContext.parallelize(1 to 10).toDF("num")
+      df.write.format("parquet").save(path)
+      // create the temporary table
+      sql(s"CREATE TEMPORARY TABLE test1 USING parquet OPTIONS (path '$path')")
```

Comment:
Lets not create a parquet file. It is easer to use `org.apache.spark.sql.sources.DDLScanSource` (test class). This is easier to setup and isn't writing to disk. This would become somthing like:

    CREATE TEMPORARY TABLE test1 
    USING org.apache.spark.sql.sources.DDLScanSource
    OPTIONS (From '1', To '10', Table 'Test')